### DISCUSSION TITLE
Association proxy eager loading

### DISCUSSION THREAD

**Discussion by dishwad** [sqlalchemy/sqlalchemy#10047] [Usage Questions]
Is it possible to do eager loading with association proxies? I've tried everything but I keep getting this error:
`sqlalchemy.exc.NoInspectionAvailable: No inspection system is available for object of type <class 'sqlalchemy.ext.associationproxy.ObjectAssociationProxyInstance'>`

Here's my query:
```python
stmt = (
    select(Account)
    .join(UserAccounts, Account.id == UserAccounts.account_id)
    .join(User, User.id == UserAccounts.user_id)
    .outerjoin(AccountProducts, Account.id == AccountProducts.account_id)
    .outerjoin(Product, Product.id == AccountProducts.product_id)
    .filter(User.id == user_id)
    .options(
        joinedload(Account.products),
    )
)
accounts = db.execute(stmt).scalars().all()
```
I'm trying to eager load `products` in my `Account` model:
```python
products: AssociationProxy[list["Product"]] = association_proxy(
    "account_products", "product"
)
```

**Comment by zzzeek** [comment]
you need to break it into separate joinedload calls

`joinedload(Account.account_products).joinedload(Product.product)`

There is a way to do the above generically but it's not wrapped into a library function right now and it would be a little ugly to roll on your end:

```py
def eagerload_prox(loader, prox):
    # given eagerload_prox(joinedload, User.keywords)

    # User.user_keyword_associations
    first_attribute =getattr(prox.owning_class, prox.target_collection)

    # joinedload(User.keywords)
    step1 = loader(first_attribute)

    # UserKeywordAssociation
    middle_class = first_attribute.property.mapper.class_

    # UserKeywordAssociation.keyword
    second_attribute = getattr(middle_class, prox.value_attr)

    # joinedload().joinedload
    eagerload_callable = getattr(step1, loader.__name__)

    # joinedload().joinedload(UserKeywordAssociation.keyword)
    step2 = eagerload_callable(second_attribute)

    return step2

```

**Reply by md384** [reply]
The separate loads did not work for me (got a `sqlalchemy.exc.ArgumentError: expected ORM mapped attribute for loader strategy argument` error), but the `eagerload_prox` function did. Perhaps this should be added to the library.
